generic/servicelog.py

```diff
@@ -0,0 +1,280 @@
+        cmd_result = process.run(cmd, ignore_status=True, sudo=True,
+                                 shell=True)
+        if cmd_result.exit_status != 0:
+            self.is_fail += 1
+        return
+
+    def setUp(self):
+        if "ppc" not in os.uname()[4]:
+            self.skip("supported only on Power platform")
+        sm = SoftwareManager()
+        if 'PowerNV' in open('/proc/cpuinfo', 'r').read():
+            self.skip("servicelog: is not supported on the PowerNV platform")
+        for package in ("servicelog", "ppc64-diag"):
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error("Fail to install %s required for this"
+                           " test." % package)
+
+    def test_servicelog(self):
+        """
+        The test checks servicelog commands.
+        """
+        rtas_events = ["v6_fru_replacement", "v6_memory_info", "v6_power_error",
+                       "v6_power_error2", "v3_predictive_cpu_failure",
+                       "v6_predictive_cpu_failure", "v6_fw_predictive_error",
+                       "v4_io_bus_failure", "v6_io_sub_error"]
+        for event in rtas_events:
+            self.log.info("Starting test scenario for %s" % event)
+            self.log.info("1 - Cleaning servicelog...")
+            self.run_cmd("servicelog_manage --truncate notify --force")
+            self.log.info("2 - Injecting event %s" % event)
+            self.run_cmd("rtas_errd -d -f %s/%s" % (self.EVENTS_PATH, event))
+            self.log.info("3 - Checking if service log does return ppc64_rtas events")
+            self.run_cmd("servicelog --type=ppc64_rtas")
+            inject = process.system_output("servicelog --type=all | "
+                                           "grep 'PPC64 Platform Event' | cut -d':' -f1", shell=True)
+            if "PPC64 Platform Event" == inject:
+                self.is_fail += 1
+                self.log.info("Event %s error: Event generated is not a "
+                              "'PPC64 Platform Event'" % event)
+            self.log.info("5 - Checking if the event was dumped to /var/log/platform")
+            self.run_cmd("cat /var/log/platform")
+            self.run_cmd("servicelog --help")
+            self.run_cmd("servicelog --version")
+            self.run_cmd("servicelog --query='id=1'")
+            self.run_cmd("servicelog --id=1")
+            self.run_cmd("servicelog --query='severity>=$WARNING AND closed=0'")
+            self.run_cmd("servicelog --serviceable=yes")
+            self.run_cmd("servicelog --serviceable=no")
+            self.run_cmd("servicelog --serviceable=all")
+            self.run_cmd("servicelog --query='serviceable=1 AND closed=0'")
+            self.run_cmd("servicelog --repair_action=yes")
+            self.run_cmd("servicelog --repair_action=no")
+            self.run_cmd("servicelog --repair_action=all")
+            self.run_cmd("servicelog --repair_action=yes")
```

Comment:
we have `self.run_cmd("servicelog --repair_action=yes")` 2 times, should we have run twice, also can we have a list of cmds and use for loop,
```
service_log_cmds = ["--help", "--version", ..]
for cmd in service_log_cmds:
    self.run_cmd("servicelog %s" % cmd)
```